### PR TITLE
SQL: Use SDK sqlutil package refactor

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Uses https://github.com/grafana/grafana-plugin-sdk-go/pull/330.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by marefr** [comment]
Some problems left running the integration tests locally.

Postgres:
```
=== RUN   TestPostgres/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_bigint_null_value_should_return_nil
    postgres_test.go:1126: 
                Error Trace:    postgres_test.go:1126
                Error:          Received unexpected error:
                                convert frame from rows error: sql: Scan error on column index 0, name "time": converting NULL to int64 is unsupported
                Test:           TestPostgres/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_bigint_null_value_should_return_nil
=== RUN   TestPostgres/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_timestamp_null_value_should_return_nil
    postgres_test.go:1156: 
                Error Trace:    postgres_test.go:1156
                Error:          Received unexpected error:
                                convert frame from rows error: sql: Scan error on column index 0, name "time": unsupported Scan, storing driver.Value type <nil> into type *time.Time
                Test:           TestPostgres/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_timestamp_null_value_should_return_nil
--- FAIL: TestPostgres (0.02s)
```

MySQL:
```
=== RUN   TestMySQL/Given_a_table_with_different_native_data_types/Query_with_Table_format_should_map_MySQL_column_types_to_Go_types                                                           
    mysql_test.go:141:                                                                                                                                                                         
                Error Trace:    mysql_test.go:141                                                                                                                                              
                Error:          Received unexpected error:                                                                                                                                     
                                convert frame from rows error: sql: Scan error on column index 27, name "aintnull": converting NULL to string is unsupported                                   
                Test:           TestMySQL/Given_a_table_with_different_native_data_types/Query_with_Table_format_should_map_MySQL_column_types_to_Go_types 
=== RUN   TestMySQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_unsigned_integer_null_value_should_return_nil
    mysql_test.go:1054: 
                Error Trace:    mysql_test.go:1054
                Error:          Received unexpected error:
                                convert frame from rows error: sql: Scan error on column index 0, name "time_sec": converting NULL to string is unsupported
                Test:           TestMySQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_unsigned_integer_null_value_should_return_nil
=== RUN   TestMySQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_DATETIME_null_value_should_return_nil
    mysql_test.go:1084: 
                Error Trace:    mysql_test.go:1084
                Error:          Received unexpected error:
                                convert frame from rows error: sql: Scan error on column index 0, name "time_sec": converting NULL to string is unsupported
                Test:           TestMySQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_DATETIME_null_value_should_return_nil
--- FAIL: TestMySQL (0.02s)
    --- FAIL: TestMySQL/Given_a_table_with_different_native_data_types (0.00s)
        --- FAIL: TestMySQL/Given_a_table_with_different_native_data_types/Query_with_Table_format_should_map_MySQL_column_types_to_Go_types (0.00s)
```

MSSQL:
```
--- FAIL: TestMSSQL (0.51s)                                                                                                                                                                    
    --- FAIL: TestMSSQL/Given_a_table_with_event_data (0.03s)                                                                                                                                  
        --- FAIL: TestMSSQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_bigint_null_value_should_return_nil (0.00s)                              
            mssql_test.go:1167:                                                                                                                                                                
                        Error Trace:    mssql_test.go:1167                                                                                                                                     
                        Error:          Received unexpected error:                                                                                                                             
                                        convert frame from rows error: sql: Scan error on column index 0, name "time": converting NULL to int64 is unsupported                                 
                        Test:           TestMSSQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_bigint_null_value_should_return_nil                
        --- FAIL: TestMSSQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_datetime_null_value_should_return_nil (0.00s)                            
            mssql_test.go:1197:                                                                                                                                                                
                        Error Trace:    mssql_test.go:1197                                                                                                                                     
                        Error:          Received unexpected error:                                                                                                                             
                                        convert frame from rows error: sql: Scan error on column index 0, name "time": unsupported Scan, storing driver.Value type <nil> into type *time.Time  
                        Test:           TestMSSQL/Given_a_table_with_event_data/When_doing_an_annotation_query_with_a_time_column_holding_a_datetime_null_value_should_return_nil              
FAIL                                                                                                                                                                                           
FAIL    github.com/grafana/grafana/pkg/tsdb/mssql       0.713s                                                                                                                                 
FAIL                    
```

File: pkg/tsdb/mysql/mysql_test.go [hunk 1]
```diff
@@ -150,11 +150,11 @@ func TestMySQL(t *testing.T) {
 			require.Equal(t, "def", *frameOne.Fields[2].At(0).(*string))
 			require.Equal(t, int32(1), frameOne.Fields[3].At(0).(int32))
 			require.Equal(t, int16(10), frameOne.Fields[4].At(0).(int16))
-			require.Equal(t, int64(100), frameOne.Fields[5].At(0).(int64))
-			require.Equal(t, int32(1420070400), frameOne.Fields[6].At(0).(int32))
+			require.Equal(t, int64(100), *(frameOne.Fields[5].At(0).(*int64)))
+			require.Equal(t, int64(1420070400), *(frameOne.Fields[6].At(0).(*int64)))
```
Comment (kminehart) [inline pkg/tsdb/mysql/mysql_test.go:154]: If you look at the converter for this test, the converter returns a `*int64`, and has a `OutputFieldType` of `NullableInt64`.

File: pkg/tsdb/mysql/mysql_test.go [hunk 2]
```diff
@@ -150,11 +150,11 @@ func TestMySQL(t *testing.T) {
 			require.Equal(t, "def", *frameOne.Fields[2].At(0).(*string))
 			require.Equal(t, int32(1), frameOne.Fields[3].At(0).(int32))
 			require.Equal(t, int16(10), frameOne.Fields[4].At(0).(int16))
-			require.Equal(t, int64(100), frameOne.Fields[5].At(0).(int64))
-			require.Equal(t, int32(1420070400), frameOne.Fields[6].At(0).(int32))
+			require.Equal(t, int64(100), *(frameOne.Fields[5].At(0).(*int64)))
```
Comment (kminehart) [inline pkg/tsdb/mysql/mysql_test.go:153]: If you look at the converter for this test, the converter returns a `*int64`, and has a `OutputFieldType` of `NullableInt64`.

File: pkg/tsdb/mysql/mysql_test.go [hunk 3]
```diff
@@ -150,11 +150,11 @@ func TestMySQL(t *testing.T) {
...
+			require.Equal(t, int64(1420070400), *(frameOne.Fields[6].At(0).(*int64)))
 			require.Equal(t, 1.11, *frameOne.Fields[7].At(0).(*float64))
 			require.Equal(t, 2.22, *frameOne.Fields[8].At(0).(*float64))
-			require.Equal(t, float32(3.33), frameOne.Fields[9].At(0).(float32))
+			require.Equal(t, float64(3.33), *(frameOne.Fields[9].At(0).(*float64)))
```
Comment (kminehart) [inline pkg/tsdb/mysql/mysql_test.go:157]: If you look at the converter for this test, the converter returns a `*int64`, and has a `OutputFieldType` of `NullableInt64`.

File: pkg/tsdb/postgres/postgres_test.go [hunk 4]
```diff
@@ -254,7 +254,7 @@ func TestPostgres(t *testing.T) {
 			require.Len(t, frames, 1)
 			require.Len(t, frames[0].Fields, 17)
 
-			require.Equal(t, int16(1), *frames[0].Fields[0].At(0).(*int16))
+			require.Equal(t, int16(1), frames[0].Fields[0].At(0).(int16))
```
Comment (kminehart) [inline pkg/tsdb/postgres/postgres_test.go:257]: For this one, this column is not nullable, so the int16s do not end up as pointers anymore.

**Comment by kminehart** [comment]
Not sure about these build failures. They don't seem relevant. What do you think?

**Comment by marefr** [comment]
No, think they are due to some other sdk changes part of sdk version not yet used by grafana. @FZambia has tagged v0.93.0 of the sdk and are planning to upgrafe grafana with this tomorrow which should resolve these errors.

**Review by kminehart — APPROVED** [review]

**Pull request merged**